object_tracker/object_tracker.go

```diff
@@ -122,6 +122,7 @@ func newTracker(ctx context.Context, deps resource.Dependencies, conf resource.C
 	}, func() {
 		t.cancelFunc()
 		stream.Close(t.cancelContext)
+		ctx.Done()
```

Comment:
You don't need to do this. This context is for building the module in the first place. You don't need to keep it around. It will become irrelevant once `newTracker` exits

Reply:
Also, ctx.Done() is a `channel` type, so I'm not sure if this syntax is correct either (are you trying to receive a value for this channel or wait until it's done? You would to use `<-ctx.Done()` for that